File: jambeth/jambeth-cache/src/main/java/de/osthus/ambeth/util/IPrioMembersProvider.java
package de.osthus.ambeth.util;

import de.osthus.ambeth.collections.ArrayList;
import de.osthus.ambeth.collections.ILinkedMap;
import de.osthus.ambeth.collections.IdentityLinkedSet;
import de.osthus.ambeth.metadata.Member;

public interface IPrioMembersProvider
{
	IdentityLinkedSet<Member> getPrioMembers(ILinkedMap<Class<?>, PrefetchPath[]> entityTypeToPrefetchPath, ArrayList<PrefetchCommand> pendingPrefetchCommands,
			MergePrefetchPathsCache mergePrefetchPathsCache);
}

File: ambeth/Ambeth.Cache/ambeth/util/IPrioMembersProvider.cs
using De.Osthus.Ambeth.Collections;
using De.Osthus.Ambeth.Metadata;
using System;
using System.Collections.Generic;

namespace De.Osthus.Ambeth.Util
{
	public interface IPrioMembersProvider
	{
		IdentityLinkedSet<Member> GetPrioMembers(ILinkedMap<Type, PrefetchPath[]> entityTypeToPrefetchPath, List<PrefetchCommand> pendingPrefetchCommands,
				MergePrefetchPathsCache mergePrefetchPathsCache);
	}
}